tools/testdata-scripts/src/bin/make-testdata.rs

```diff
@@ -68,13 +48,7 @@ fn main() {
 
     icu_datagen::datagen(
         Some(&icu_testdata::locales()),
-        &icu_datagen::all_keys_with_experimental()
-            .into_iter()
-            .filter(|k| !IGNORED_KEYS.contains(&&*k.path()))
-            .chain(core::iter::once(
-                icu_provider::hello_world::HelloWorldV1Marker::KEY,
-            ))
-            .collect::<Vec<_>>(),
+        &icu_datagen::all_keys_with_experimental(),
```

Comment:
Nit: I'd prefer to keep the hello_world key and revert the changes related to that. It has been a useful inclusion, such as the ability to use it in docstests. It's not clear that it's an improvement that we need to use some other component like icu_list or pull data from the HelloWorldProvider instead of the testdata provider.

Reply:
I'll split it out to discuss further